FIAT/recursive_points.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2015 Imperial College London and others.
+#
+# This file is part of FIAT (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+#
+# Written by Pablo D. Brubeck (brubeck@protonmail.com), 2023
+
+import numpy
+
+"""
+@article{isaac2020recursive,
+  title={Recursive, parameter-free, explicitly defined interpolation nodes for simplices},
+  author={Isaac, Tobin},
+  journal={SIAM Journal on Scientific Computing},
+  volume={42},
+  number={6},
+  pages={A4046--A4062},
+  year={2020},
+  publisher={SIAM}
+}
+"""
+
+
+def multiindex_equal(d, isum, imin=0):
+    """A generator for d-tuple multi-indices whose sum is isum and minimum is imin.
+    """
+    if d <= 0:
+        return
+    imax = isum - (d - 1) * imin
+    if imax < imin:
+        return
+    for i in range(imin, imax):
+        for a in multiindex_equal(d - 1, isum - i, imin=imin):
+            yield a + (i,)
+    yield (imin,) * (d - 1) + (imax,)
+
+
+class RecursivePointSet(object):
```

Comment:
Do we want to re-implement Toby's work here or just create a dependency?  If we did, we would have access to all the other point sets he implements without re-doing them ourselves.